webserver/src/main.rs

```diff
@@ -0,0 +1,417 @@
+        team_stats
+            .into_iter()
+            .map(|(t, s)| (format!("{}, {}", t[0], t[1]), s)),
+    );
+    final_player_stats.sort_by(|a, b| b.pointrate.partial_cmp(&a.pointrate).unwrap());
+    final_team_stats.sort_by(|a, b| b.pointrate.partial_cmp(&a.pointrate).unwrap());
+
+    let leaderboard = |final_stats: Vec<FinalStats>| {
+        final_stats
+            .into_iter()
+            .map(|s| {
+                rsx! {
+                    <tr>
+                        <td>{s.name}</td>
+                        <td>{format!("{:.3}", s.pointrate)}</td>
+                        <td>{s.points}</td>
+                        <td>{s.games}</td>
+                        <td>{s.wins}</td>
+                        <td>{s.losses}</td>
+                        <td>{s.draws}</td>
+                    </tr>
+                }
+            })
+            .collect::<Vec<_>>()
+    };
+
+    let h: String = html! {
+        <html>
+            <style>
+                {STYLESHEET}
+            </style>
+        <head>
+        </head>
+        <body>
+          <table>
+            <p>{"Player Leaderboard"}</p>
+            <tr>
+                <th>{"Player"}</th>
+                <th>{"Pointrate"}</th>
+                <th>{"Points"}</th>
+                <th>{"Games"}</th>
+                <th>{"Wins"}</th>
+                <th>{"Losses"}</th>
+                <th>{"Draws"}</th>
+            </tr>
+            {leaderboard(final_player_stats)}
+          </table>
+          <table>
+            <p>{"Team Leaderboard"}</p>
+            <tr>
+                <th>{"Team"}</th>
+                <th>{"Pointrate"}</th>
+                <th>{"Points"}</th>
+                <th>{"Games"}</th>
+                <th>{"Wins"}</th>
+                <th>{"Losses"}</th>
+                <th>{"Draws"}</th>
+            </tr>
+            {leaderboard(final_team_stats)}
+          </table>
+        </body>
+        </html>
+    };
+    let mut resp = Response::new(StatusCode::Ok);
+    resp.set_content_type(Mime::from("text/html; charset=UTF-8"));
+    resp.set_body(h);
+    Ok(resp)
+}
+
+fn process_stats<I: Iterator<Item = (String, RawStats)>>(raw_stats: I) -> Vec<FinalStats> {
+    raw_stats
+        .map(|(name, s)| {
+            let games = s.draws + s.wins + s.losses;
+            let points = s.wins as f64 + 0.5 * s.draws as f64;
+            FinalStats {
+                name,
+                games,
+                points,
+                draws: s.draws,
+                wins: s.wins,
+                losses: s.losses,
+                pointrate: points / games as f64,
+            }
+        })
+        .collect()
+}
+
+fn format_timestamp_date_and_time(maybe_ts: Option<i64>) -> Option<(String, String)> {
+    let ts = maybe_ts?;
+    let datetime = OffsetDateTime::from_unix_timestamp(ts).ok()?;
+    let date = datetime
+        .format(
+            &time::format_description::parse("[year]-[month]-[day], [weekday repr:short]").ok()?,
```

Comment:
Would `[format_description!](https://docs.rs/time/latest/time/macros/macro.format_description.html)` work here? The more compile-time check the better, right? :)

Reply:
Absolutely, that's great!